lib/screens/scan_screen.dart

```diff
@@ -0,0 +1,166 @@
+import 'dart:convert';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:scan/scan.dart';
+import 'package:http/http.dart' as http;
+import 'package:image_picker/image_picker.dart';
+
+// ignore: must_be_immutable
+class ScanScreen extends StatefulWidget {
+  dynamic data;
```

Comment:
issue: Using dynamic parameters makes this class vulnerable to bugs since you can pass any type of value (whether it's null or non-null)